CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+cmake_minimum_required(VERSION 3.22)
```

Comment:
A good practice is to use cmake_minimum_required as low as possible. We do not use any advance cmake features (which were introduced recently) + people with older distros will not be able to compile the code with cmake available in apt. They will need either edit CMakeLists.txt or clone/build/install cmake from sources.

Reply:
@kawapa Which cmake version would you use?

Reply:
3.14 Tested with https://github.com/nlohmann/cmake_min_version

Reply:
I'm not convinced to that tool. When reviewing it, I took 2.15 and it worked.

Reply:
Honestly, it's my first time using this tool, but in the project, I saw the 'FetchContent_MakeAvailable' command. According to the documentation, it makes sense.
https://cmake.org/cmake/help/latest/module/FetchContent.html#command:fetchcontent_makeavailable